stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/images/AttachmentGalleryActivity.kt

```diff
@@ -99,16 +115,30 @@ public class AttachmentGalleryActivity : AppCompatActivity() {
         private const val EXTRA_KEY_USER_NAME = "extra_key_user_name"
         private const val EXTRA_KEY_TIME = "extra_key_time"
 
+        private lateinit var attachmentShowInChatOptionHandler: AttachmentShowInChatOptionHandler
+        private lateinit var attachmentReplyOptionHandler: AttachmentReplyOptionHandler
+        private lateinit var attachmentDownloadOptionHandler: AttachmentDownloadOptionHandler
+        private lateinit var attachmentDeleteOptionHandler: AttachmentDeleteOptionHandler
```

Comment:
These are static fields in the companion, is this intentional?

Reply:
I couldn't find a better way to inject objects of anonymous classes into Activity.

Reply:
Any idea how to improve this?

Reply:
For inter-Activity communication, perhaps we could use the [Activity Result APIs](https://developer.android.com/training/basics/intents/result)? That's what we do for photo / file selections as well.

Reply:
Looks good, thanks. I'll refactor static handlers this way.